zb-db/src/main/java/io/camunda/zeebe/db/impl/rocksdb/transaction/ForeignKeyChecker.java

```diff
@@ -21,9 +22,12 @@
 public final class ForeignKeyChecker {
   private final ZeebeTransactionDb<?> transactionDb;
   private final ExpandableArrayBuffer keyBuffer = new ExpandableArrayBuffer();
+  private final boolean enabled;
 
-  public ForeignKeyChecker(final ZeebeTransactionDb<?> transactionDb) {
+  public ForeignKeyChecker(
+      final ZeebeTransactionDb<?> transactionDb, final ConsistencyChecksSettings settings) {
```

Comment:
:wrench: Probably would be sufficient to pass in only the flag. Or do you think we will have more feature flags related to this class?

Reply:
I'm leaving this as is. You are right that the single boolean would be enough but this way we have a bit more flexibility for more configuration :shrug: